### PR TITLE
Do not ask a UI question on metrics screen

### LINKED ISSUE #2095: showmetrics interferes with start-up error prompts
When -showmetrics is active, the "Do you want to rebuild the block database now?" prompt does not work (see https://forum.z.cash/t/zcash4win-beta-is-out/14155/13 ).

### PR DESCRIPTION
**Pull request by str4d**
Based on bitcoin/bitcoin#8257.

Closes #2095.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -1341,8 +1341,9 @@ bool AppInit2(boost::thread_group& threadGroup, CScheduler& scheduler)
         if (!fLoaded) {
             // first suggest a reindex
             if (!fReset) {
-                bool fRet = uiInterface.ThreadSafeMessageBox(
+                bool fRet = uiInterface.ThreadSafeQuestion(
                     strLoadError + ".\n\n" + _("Do you want to rebuild the block database now?"),
+                    strLoadError + ".\nPlease restart with -reindex or -reindex-chainstate to recover.",
```
Comment (daira) [inline src/init.cpp]: We don't have ``-reindex-chainstate``. That was added in upstream 0.14.0: https://github.com/bitcoin/bitcoin/commit/d3d75479115bc3480f163df774ee9dd2f8bd9f54

Comment (daira) [inline src/init.cpp]: Oops, I didn't spot that you'd changed this in a subsequent commit.

**Comment by daira** [comment]
utACK.

**Comment by arcalinea** [comment]
Tested, can confirm the metrics screen no longer asks a question when block db is corrupted.

ACK

jay jay-debian -zcash-zcash_078

**Comment by arcalinea** [comment]
@zkbot r+

**Comment by daira** [comment]
@zkbot retry

**Pull request merged**